cmd/helm/downloader/manager.go

```diff
@@ -480,3 +512,38 @@ func writeLock(chartpath string, lock *chartutil.RequirementsLock) error {
 	dest := filepath.Join(chartpath, "requirements.lock")
 	return ioutil.WriteFile(dest, data, 0644)
 }
+
+// archive a dep chart from local directory and save it into charts/
+func tarFromLocalDir(chartpath string, name string, repo string, version string) (string, error) {
+	destPath := filepath.Join(chartpath, "charts")
+	origPath, err := filepath.Abs(repo[7:])
+	if err != nil {
+		return "", err
+	}
+
+	if _, err = os.Stat(origPath); os.IsNotExist(err) {
+		return "", fmt.Errorf("directory %s not found: %s", origPath, err)
+	}
+
+	ch, err := chartutil.LoadDir(origPath)
+	if err != nil {
+		return "", err
+	}
+
+	constraint, err := semver.NewConstraint(version)
```

Comment:
Glad you did the SemVer constraint stuff here. I think this will be really helpful to users.